src/logic/map_objects/tribes/soldier.cc

```diff
@@ -1747,13 +1747,21 @@ void Soldier::naval_invasion_update(Game& game, State& state) {
 		}
 	}
 
+	// The player doesn't have to build the port exactly where we envision it.
+	if (map.find_reachable_immovables(
+	       game, Area<FCoords>(get_position(), kPortSpaceGeneralAreaRadius), nullptr, checkstep,
+	       FindFlagWithPlayersWarehouse(*get_owner())) > 0) {
```

Comment:
Why was the check for a port at the invasion spot kept? Doesn't this cover that case too?

Reply:
If the soldier is standing right on the port when it's built, he might be unable to find a path anywhere when simply popping the task. So in this case he still needs to be incorporated. Besides it's nicer and also faster to return directly to the port instead of wandering around to a random flag first.

Reply:
OK.